### PR TITLE
Initial test for shot based animation workflow with custom code-less schema

### PR DESCRIPTION
**Pull request by kxl-adsk**

### CODE REVIEW

**Comment by kxl-adsk** [comment]
This is really a test to validate that we can write a complete shot-based animation workflow on top of a custom codeless schema. All python bindings worked out great!

File: test/lib/mayaUsd/fileio/testCustomRig.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        super(customRigPrimUpdater, self).__init__(*args, **kwargs)
+
+    def editAsMaya(self,context):
+        return super(customRigPrimUpdater, self).editAsMaya(context)
+
+    def discardEdits(self,context):
+        return super(customRigPrimUpdater, self).discardEdits(context)
+
+    def pushCopySpecs(self, srcStage, srcLayer, srcSdfPath, dstStage, dstLayer, dstSdfPath):
+        #return super(customRigPrimUpdater, self).pushCopySpecs(srcStage, srcLayer, srcSdfPath, dstStage, dstLayer, dstSdfPath)
```
Comment (ppt-adsk) [inline test/lib/mayaUsd/fileio/testCustomRig.py:66]: Commented-out code for reference or experimentation purposes?

File: test/lib/mayaUsd/fileio/testCustomRig.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        )
+        
+        bobUfePathStr = "{},/world/anim/bob".format(proxyShape)
+        
+        # Pull the object for editing in Maya
+        self.assertTrue(mayaUsdLib.PrimUpdaterManager.editAsMaya(bobUfePathStr))
+        
+        # Retrieve pulled object
+        bobMayaPathStr = cmds.ls(sl=True)[0]
+        #bobMayaItem = ufe.GlobalSelection.get().front()
```
Comment (ppt-adsk) [inline test/lib/mayaUsd/fileio/testCustomRig.py:176]: This commented-out code doesn't seem to be useful.

File: test/lib/mayaUsd/fileio/testCustomRig.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+        
+        # Animate the cubes
+        cmds.playbackOptions( minTime=0, maxTime=10 )
+        cmds.setKeyframe( "bob|pCube1.ry", time=1.0, value=0 )
+        cmds.setKeyframe( "bob|pCube1.ry", time=10.0, value=100 )
+        cmds.setKeyframe( "bob|pCube2.ty", time=1.0, value=0 )
+        cmds.setKeyframe( "bob|pCube2.ty", time=10.0, value=10 )
+        
+        # Push the animation back to USD. This will use a custom logic that will write it to a new prim
+        #bobMayaItemPathStr = ufe.PathString.string(bobMayaItem.path())
```
Comment (ppt-adsk) [inline test/lib/mayaUsd/fileio/testCustomRig.py:192]: No longer useful, it seems.

**Review by ppt-adsk — APPROVED** [review]
Nice!  This is a clean demonstration for customers of how they can customize our workflow.

**Review by boudrey — APPROVED** [review]

**Review by ppt-adsk — APPROVED** [review]

**Pull request merged**